src/Controller/ApplenewsPreviewController.php

```diff
@@ -127,42 +126,58 @@ protected function getDataArray(EntityInterface $entity, array $context) {
    *
    * @param int $entity_id
    *   Entity ID.
-   * @param array $entity_ids
-   *   An array of entity IDs.
    * @param string $filename
    *   String filename.
    * @param string $entity_archive
    *   String path.
    * @param array $data
    *   An array of article dta.
    *
-   * @return int|null|string
+   * @return null|string
```

Comment:
I'm still unsure about this return. Is this saying a URL is returned if something goes wrong? Wouldn't it be an array of URLs if it's a group of entities? Rather than catch the exception, why not let it bubble and then catch in a controller or form or something?

Reply:
Makes so much more sense now! Thanks.